### PR TITLE
Make it clearer that releases are non Apache releases

### PR DESCRIPTION
**Pull request by justinmclean**
Make it more apparent that releases are non-Apache releases and improve English.

### CODE REVIEW

File: amoro-site/content/download.md [hunk 1]
```diff
@@ -6,17 +6,20 @@ disableSidebar: true
...
...ttps://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-0.6.1-bin.zip) <br> - [AMS(hadoop2)](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-0.6.1-bin.hadoop2.zip) | - [Flink 1.15 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-flink-runtime-1.15-0.6.1.jar) <br> - [Flink 1.16 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-flink-runtime-1.16-0.6.1.jar) <br> - [Flink 1.17 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-flink-runtime-1.17-0.6.1.jar) <br> | - [Spark 3.1 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-spark-3.1-runtime-0.6.1.jar)<br> - [Spark 3.2 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-spark-3.2-runtime-0.6.1.jar)<br> - [Spark 3.3 Runtime Jar](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/amoro-spark-3.3-runtime-0.6.1.jar) | [Trino Connector](https://github.com/apache/incubator-amoro/releases/download/v0.6.1/trino-amoro-0.6.1.tar.gz) | [release note](https://github.com/apache/incubator-amoro/releases/tag/v0.6.1) |
```
Comment (zhoujinsong) [inline amoro-site/content/download.md:21]: Maybe we can drop the `(non-Apache)` description here for 0.6.1.

**Review by zhoujinsong — COMMENTED** [review]
@justinmclean 
Thanks a lot for your contribution!
The non-apache releases are much clearer. I left a small suggestion, please take a look.

File: amoro-site/content/download.md (hunk 1 above)
Comment (justinmclean) [reply amoro-site/content/download.md:21]: No, we cannot, as it is not an ASF release. For it to be an ASF release, it needs to go through the ASF release process and be voted on by the Amoro PPMc and the Incubator PMC.

Comment (zhoujinsong) [reply amoro-site/content/download.md:21]: I agree that version 0.6.1 is also a non-apache release and it is under the title: Non-Apache releases.

What I want to say is that in the release table, in order to maintain consistency with other versions' formats, we only fill in the version names in the "version" column and remove any additional descriptions.

However, we can improve the format in another PR, this PR is totally okay.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**